packages/frontend-api/src/Component/Arguments/ProductSearchPaginatorArgumentsBuilder.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Component\Arguments;
+
+class ProductSearchPaginatorArgumentsBuilder extends ProductPaginatorArgumentsBuilder
+{
+    /**
+     * @param array $config
+     * @return array
+     */
+    public function toMappingDefinition(array $config): array
+    {
+        $this->checkMandatoryFields($config);
+
+        $mappingDefinition = parent::toMappingDefinition($config);
+
+        unset($mappingDefinition['categorySlug'], $mappingDefinition['brandSlug'], $mappingDefinition['flagSlug']);
```

Comment:
IMHO this is too specific - if we added something into `ProductPaginatorArgumentsBuilder::toMappingDefinition`, it might break `ProductSearchPaginatorArgumentsBuilder`and no one might notice that. 

I suggest not extending one product builder from another but creating one more abstract class for product paginator builders that would extend the default abstract one, and then `ProductSearchPaginatorArgumentsBuilder` and `ProductPaginatorArgumentsBuilder` would extend that new abstract class, what do you think?

Reply:
thanks, so you can remove the line :slightly_smiling_face: 

```
unset($mappingDefinition['categorySlug'], $mappingDefinition['brandSlug'], $mappingDefinition['flagSlug']);
```